tests/test_read.py

```diff
@@ -411,3 +411,34 @@ def test_reading_mutually_exclusive_group():
         client_data = data.location(client)  # type: ignore
         assert_equal(client_data.required, False)
         assert_equal(client_data.group, 0)
+
+
+def test_heterogeneous_fleet():
+    """
+    Tests that reading an instance with heterogeneous fleet works correctly.
+    """
+    data = read("data/OkSmallHeterogeneousFleet.txt")
+    assert_equal(data.num_vehicles, 4)
+    assert_equal(data.num_vehicle_types, 3)
+
+    num_vehicles = [2, 1, 1]
+    capacities = [1, 2, 3]
+    fixed_cost = [100, 200, 300]
+    variable_cost = [1, 2, 2]
+    names = ["0,1", "2", "3"]
+
+    for idx, vehicle_type in enumerate(data.vehicle_types()):
+        assert_equal(vehicle_type.num_available, num_vehicles[idx])
+        assert_equal(vehicle_type.capacity, capacities[idx])
+        assert_equal(vehicle_type.fixed_cost, fixed_cost[idx])
+        assert_equal(vehicle_type.unit_distance_cost, variable_cost[idx])
+        assert_equal(vehicle_type.name, names[idx])
+
+
+def test_raises_num_vehicles_not_enough_vehicle_data():
+    """
+    Tests that ``read`` raises when vehicle data sections do not have the same
+    number of elements as the number of vehicles in the instance.
+    """
+    with assert_raises(ValueError):
+        read("data/MoreVehiclesThanSectionData.txt")
```

Comment:
```suggestion
def test_raises_num_vehicles_not_enough_vehicle_data():
    """
    Tests that ``read`` raises when vehicle data sections do not have the same
    number of elements as the number of vehicles in the instance.
    """
    with assert_raises(ValueError):
        read("data/MoreVehiclesThanSectionData.txt")
```
can we add this to the parameterised `test_raises_invalid_file` all the way at the top of the file instead?